File: src/main/java/org/cmjava2023/ClassfileModel.java
package org.cmjava2023;

public record ClassfileModel(
    short sizeOfConstantPool,
    byte[] constantPool,                    // Array of own class?
    short classAccessModifiers,
    short constantPoolIndex,
    short superClassIndexInConstantPool,
    short numberOfInterfaces,
    byte[] interfaceDefinitions,            // Array of own class?
    short numberOfFields,
    byte[] fieldDefinitions,                // Array of own class?
    short numberOfMethods,
    byte[] methodDefinitions,               // Array of own class?
    short attributesCount,
    byte[] attributeDefinitions             // Array of own class?
){}

File: src/main/kotlin/org/cmjava2023/ClassfileModel.kt
package org.cmjava2023

data class ClassfileModel(
    val sizeOfConstantPool: Short,
    val constantPool: List<Byte>,  // Array of own class?
    val classAccessModifiers: Short,
    val constantPoolIndex: Short,
    val superClassIndexInConstantPool: Short,
    val numberOfInterfaces: Short,
    val interfaceDefinitions: List<Byte>,  // Array of own class?
    val numberOfFields: Short,
    val fieldDefinitions: List<Byte>,  // Array of own class?
    val numberOfMethods: Short,
    val methodDefinitions: List<Byte>,  // Array of own class?
    val attributesCount: Short,
    val attributeDefinitions: List<Byte>
)
